src/main/scala/URModel.scala

```diff
@@ -17,184 +17,106 @@
 
 package org.template
 
-import java.util.Date
-
 import grizzled.slf4j.Logger
-
-import io.prediction.data.storage.PropertyMap
+import io.prediction.data.storage.DataMap
 import org.apache.mahout.math.indexeddataset.IndexedDataset
-import org.apache.spark.rdd.RDD
 import org.apache.mahout.sparkbindings.indexeddataset.IndexedDatasetSpark
+import org.apache.spark.SparkContext
+import org.apache.spark.rdd.RDD
 import org.joda.time.DateTime
 import org.json4s.JsonAST.JArray
 import org.json4s._
-import org.template.conversions.IndexedDatasetConversions
-import org.elasticsearch.spark._
-import org.apache.spark.SparkContext
-
+import org.template.conversions.{ IndexedDatasetConversions, ItemID, ItemProps }
 
 /** Universal Recommender models to save in ES */
 class URModel(
-    coocurrenceMatrices: Option[List[(String, IndexedDataset)]],
-    fieldsRDD: Option[RDD[(String, PropertyMap)]],
-    indexName: String,
-    dateNames: Option[List[String]] = None,
-    nullModel: Boolean = false,
-    typeMappings: Option[Map[String, String]] = None, // maps fieldname that need type mapping in Elasticsearch
-    propertiesRDD: Option[RDD[collection.Map[String, Any]]] = None)
-     {
-  @transient lazy val logger = Logger[this.type]
+    coocurrenceMatrices: Seq[(ItemID, IndexedDataset)] = Seq.empty,
+    propertiesRDDs: Seq[RDD[(ItemID, ItemProps)]] = Seq.empty,
+    typeMappings: Map[String, String] = Map.empty, // maps fieldname that need type mapping in Elasticsearch
+    nullModel: Boolean = false)(implicit sc: SparkContext) {
```

Comment:
`fieldsRDD` and `propertiesRDD` now is `propertiesRDDs: Seq[RDD[(ItemID, ItemProps)]]`